### PR TITLE
net: arp: Fix dead lock caused by arp_mutex

### PR DESCRIPTION
**Pull request by MaochenWang1**
Fix the dead lock between rx_q and tx_q thread.
When tx_q thread prepares to send ARP packet, it might get the net_if_tx_lock in net_if_tx(), then in net_arp_prepare(), it will try to get the arp_mutex.
At the same time, if the rx_q thread receives an ARP reply packet, in arp_update(), it will get the arp_mutex first, and flush the packets in arp pending_queue and try to get the net_if_tx_lock. Then the dead lock occurs, two threads stuck and all the packcets can't be freed.
In arp_update(), taking the net_if_tx_lock first then taking the arp_mutex can fix this issue.

### CODE REVIEW

File: subsys/net/l2/ethernet/arp.c [hunk 1]
```diff
@@ -467,7 +467,7 @@ void net_arp_update(struct net_if *iface,
 	struct net_pkt *pkt;
```
Comment (jukkar) [inline subsys/net/l2/ethernet/arp.c:467]: Looks good, minor nit about the commit subject, please update it to
`net: arp: Fix dead lock caused by arp_mutex`

Comment (MaochenWang1) [reply subsys/net/l2/ethernet/arp.c:467]: Hi jukkar, have updated the commit title, thanks

**Review by rlubos — APPROVED** [review]

**Review by gannicus-gj — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Comment by MaochenWang1** [comment]
Hi @jukkar @rlubos 
please help merge this fix, thanks

**Comment by jukkar** [comment]
As the PR is already approved, it will be merged by someone with commit rights when they have time. I would assume during this week, so please be patient.

**Pull request merged**